lib/history/sqlite/sqlite.go

```diff
@@ -196,13 +165,38 @@ func (t *Timeline) RecordTimeline(ctx context.Context, events []*pb.TimelineEven
 		filtered = append(filtered, event)
 	}
 
-	if err = t.insertEvents(ctx, filtered); err != nil {
+	if err := t.insertEvents(ctx, filtered); err != nil {
 		return trace.Wrap(err)
 	}
 
 	return nil
 }
 
+// insertEvents inserts the provided events into the timeline.
+func (t *Timeline) insertEvents(ctx context.Context, events []*pb.TimelineEvent) error {
+	sqlExecer := newSQLExecer(t.database)
+	for _, event := range events {
+		row, err := newDataInserter(event)
```

Comment:
Really a nit, but in scenarios like this when you iterate over something and process items one-by-one, I like to split the "single item processing" into a separate method as well which makes it a bit cleaner (and potentially, more reusable):

```go
func (t *Timeline) insertEvents(ctx context.Context, events []*pb.TimelineEvent) error {
  for _, event := range events {
    if err := t.insertEvent(ctx, event); err != nil {
      return trace.Wrap(err)
    }
  }
  return nil
}

func (t *Timeline) insertEvent(ctx context.Context, event *pb.TimelineEvent) error {
  // process event
}
```